app/services/store_sp_metadata_in_session.rb

```diff
@@ -5,6 +5,8 @@ def initialize(session:, request_id:)
   end
 
   def call
+    return unless sp_request
```

Comment:
Yes, we should validate the request ID. Note that this should rarely happen since the link in the confirmation email puts the `request_id` before `confirmation_token`, and if the user only partially copies over the URL, it will fail because we validate the confirmation token. This particular error in `int` was caused by the same user who was not following the regular flow.